graphix/flow/core.py

```diff
@@ -0,0 +1,477 @@
+    ) -> XZCorrections[_M_co]:
+        """Create an `XZCorrections` instance from the XZ-corrections mappings.
+
+        Parameters
+        ----------
+        og : OpenGraph[_M_co]
+            Open graph with respect to which the corrections are defined.
+        x_corrections : Mapping[int, AbstractSet[int]] | None
+            Mapping of X-corrections: in each (`key`, `value`) pair, `key` is a measured node, and `value` is the set of nodes on which an X-correction must be applied depending on the measurement result of `key`.
+        z_corrections : Mapping[int, AbstractSet[int]] | None
+            Mapping of X-corrections: in each (`key`, `value`) pair, `key` is a measured node, and `value` is the set of nodes on which an X-correction must be applied depending on the measurement result of `key`.
+
+        Returns
+        -------
+        XZCorrections[_M_co]
+
+        Notes
+        -----
+        This method computes the partial order induced by the XZ-corrections.
+        """
+        x_corrections = x_corrections or {}
+        z_corrections = z_corrections or {}
+
+        nodes_set = set(og.graph.nodes)
+        outputs_set = frozenset(og.output_nodes)
+        non_outputs_set = nodes_set - outputs_set
+
+        if not set(x_corrections).issubset(non_outputs_set):
+            raise ValueError("Keys of input X-corrections contain non-measured nodes.")
+        if not set(z_corrections).issubset(non_outputs_set):
+            raise ValueError("Keys of input Z-corrections contain non-measured nodes.")
+
+        dag = _corrections_to_dag(x_corrections, z_corrections)
+        partial_order_layers = _dag_to_partial_order_layers(dag)
+
+        if partial_order_layers is None:
+            raise ValueError(
+                "Input XZ-corrections are not runnable since the induced directed graph contains closed loops."
+            )
+
+        # If the open graph has outputs, the first element in the output of `_dag_to_partial_order_layers(dag)` may or may not contain a subset of the output nodes.
+        if outputs_set:
+            shift = 1 if partial_order_layers[0].issubset(outputs_set) else 0
+            partial_order_layers = [outputs_set, *partial_order_layers[shift:]]
+
+        ordered_nodes = {node for layer in partial_order_layers for node in layer}
+        if not ordered_nodes.issubset(nodes_set):
+            raise ValueError("Values of input mapping contain labels which are not nodes of the input open graph.")
+
+        # We append to the last layer (first measured nodes) all the non-output nodes not involved in the corrections.
+        if unordered_nodes := frozenset(nodes_set - ordered_nodes):
+            partial_order_layers.append(unordered_nodes)
+
+        return XZCorrections(og, x_corrections, z_corrections, tuple(partial_order_layers))
+
+    def to_pattern(
+        self: XZCorrections[Measurement],
+        total_measurement_order: TotalOrder | None = None,
+    ) -> Pattern:
```

Comment:
This function returns a standardized pattern and could return `StandardizedPattern`.